src/proof/nova.rs

```diff
@@ -1912,4 +1914,39 @@ mod tests {
             false,
         );
     }
+
+    #[test]
+    fn outer_prove_begin() {
+        let mut s = Store::<Fr>::default();
+
+        let src = "(begin (emit 1) (emit 2) (emit 3))";
+
+        let expr = s.read(src).unwrap();
+        let limit = 300;
+
+        let (
+            IO {
+                expr: _result_expr,
+                env: _new_env,
+                cont: _continuation,
+            },
+            _iterations,
+            emitted,
+        ) = Evaluator::new(expr, empty_sym_env(&s), &mut s, limit).eval();
+
+        outer_prove_aux(
+            &src,
+            |store| store.num(3),
+            Status::Terminal,
+            13,
+            DEFAULT_CHUNK_FRAME_COUNT,
+            DEFAULT_CHECK_NOVA,
+            true,
+            300,
+            false,
+        );
+        assert_eq!(s.num(1), emitted[0]);
+        assert_eq!(s.num(2), emitted[1]);
+        assert_eq!(s.num(3), emitted[2]);
```

Comment:
We should check that `emitted` contains exactly three elements, also.

Or, maybe it's simpler just to construct the expected value (of emitted) and check against that. Either way, let's be explicit about checking that it is *exactly* as it should be.